assets/js/components/settings/SettingsModules.js

```diff
@@ -19,283 +19,84 @@
-					0 < activeModules.length && // If we have active modules.
-					0 === activeTab ) && // If <SettingsApp/> is on the Connected tab. TODO this could be removed after refactoring this into separate components.
-					<div className="
-						mdc-layout-grid__cell
-						mdc-layout-grid__cell--span-12
-					">
-						<Layout>
-							{ activeModules }
-						</Layout>
-					</div>
-				}
-
-				{ /* Inactive Modules */ }
-				{ 1 === activeTab && // If <SettingsApp/> is on the Add tab. TODO this could be removed after refactoring this into separate components.
-					inactiveModulesAvailable && // We have inactive modules available.
-
-					<div className="
-								mdc-layout-grid__cell
-								mdc-layout-grid__cell--span-12
-							">
-						<Layout
-							header
-							title={ __( 'Connect More Services to Gain More Insights', 'google-site-kit' ) }
-							relative
-						>
-							<div className="mdc-layout-grid">
-								<div className="mdc-layout-grid__inner">
-									{ 0 < inactiveModules.length && inactiveModules }
-								</div>
-							</div>
-							{ /* TODO: Need some input here with regards to changing state */ }
-							{ editActive && <SettingsOverlay /> }
-						</Layout>
-					</div>
-				}
-				{ 1 === activeTab && // If <SettingsApp/> is on the Add tab. TODO this could be removed after refactoring this into separate components.
-					! inactiveModulesAvailable && // If we have no active modules.
-					<div className="
-						mdc-layout-grid__cell
-						mdc-layout-grid__cell--span-12
-					">
-						<Notification
-							id="no-more-modules"
-							title={ __( 'Congrats, you’ve connected all services!', 'google-site-kit' ) }
-							description={ __( 'We’re working on adding new services to Site Kit by Google all the time, so please check back in the future.', 'google-site-kit' ) }
-							format="small"
-							smallImage={ global._googlesitekitLegacyData.admin.assetsRoot + thumbsUpImage }
-							type="win-success"
-						/>
-					</div>
-				}
-			</Fragment>
-		);
-	}
+	const toDependantModules = ( dependantModules, dependantSlug ) => {
+		const dependantModule = modulesData[ dependantSlug ];
+		return dependantModule
+			?	[ ...dependantModules, dependantModule.name ]
+			: dependantModules;
+	};
+
+	const getDependantModulesText = ( { dependants } ) =>
+		dependants && dependants.length > 0
+			? listFormat( dependants.reduce( toDependantModules, [] ) )
+			: '';
+
+	const withDependantModulesText = ( module ) => ( {
+		...module,
+		dependantModulesText: getDependantModulesText( module ),
+	} );
+
+	const byActiveNoInternals = ( active ) => ( module ) =>
+		! module.internal && active === module.active;
+
+	const getModulesByActive = ( active ) =>
+		Object.values( modulesData )
+			.filter( byActiveNoInternals( active ) )
+			.sort( ( a, b ) => a.sort - b.sort )
+			.map( withDependantModulesText );
+
+	const activeModules = getModulesByActive( true );
+	const inactiveModules = getModulesByActive( false );
```

Comment:
Nitpick, but this API is a bit tough-to-parse. How about:

```suggestion
	const getModulesByStatus = ( { active } = {} ) =>
		Object.values( modulesData )
			.filter( byActiveNoInternals( active ) )
			.sort( ( a, b ) => a.sort - b.sort )
			.map( withDependantModulesText );

	const activeModules = getModulesByStatus( { active: true } );
	const inactiveModules = getModulesByStatus( { active: false } );
```

Reply:
Life saver! I was trying to think of better naming & personally dislike passing booleans straight into functions (no context) but ended up with `getModulesByActive( { active: true } )` which seemed worse. `getModuleByStatus` naming is exactly what I was looking for, naming is hard sometimes. 😅